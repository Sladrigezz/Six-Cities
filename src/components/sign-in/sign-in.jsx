import React from 'react';
import PropTypes from 'prop-types';
import {connect, useSelector} from 'react-redux';
import {isNil} from 'ramda';

import {Link} from 'react-router-dom';
import {getResponses} from '../../selectors/data.js';
import {getIsAuthRequired} from '../../selectors/user.js';

export const SignIn = () => {
  const isAuthRequired = useSelector(state => getIsAuthRequired(state));
  const responses = useSelector(state => getResponses(state));
  const email = !isNil(responses.auth) ? responses.auth.email : ``;
  return (
    <Link to={`/favorites`} className={`header__nav-link header__nav-link--profile`}>
      <div className="header__avatar-wrapper user__avatar-wrapper">
      </div>
      <span className="header__user-name user__name">{isAuthRequired ? `Sign in` : email}</span>
    </Link>
  );
}


SignIn.propTypes = {
  responses: PropTypes.shape({
    auth: PropTypes.shape({
      email: PropTypes.string
    })
  }),
  isAuthRequired: PropTypes.bool
};

export default connect (
  (state)=>({
    isAuthRequired: getIsAuthRequired(state),
    responses: getResponses(state),
  })
)(SignIn)
