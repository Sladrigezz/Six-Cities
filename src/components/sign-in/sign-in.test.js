import React from 'react';
import renderer from 'react-test-renderer';
import {SignIn} from './sign-in.jsx';

it(`should match snapshot`, () => {
  const props = {
    updateFieldValue: jest.fn(),
    login: jest.fn(),
  };

  const wrapper = renderer.create(<SignIn {...props} />).toJSON();

  expect(wrapper).toMatchSnapshot();
});
