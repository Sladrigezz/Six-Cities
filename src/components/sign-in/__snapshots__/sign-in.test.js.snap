// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapShot 1`] = `
<a
  className="header__nav-link header__nav-link--profile"
  href="/favorites"
  onClick={[Function]}
>
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
  />
  <span
    className="header__user-name user__name"
  >
    aaa@bbb.cz
  </span>
</a>
`;
